File: packages/snow_draw_engine/lib/draw/input/policies/drag_threshold_policy.dart
import '../../types/draw_point.dart';

/// Returns whether movement from [from] to [to] reaches [threshold].
///
/// Non-positive thresholds are treated as immediate drag start.
bool hasReachedDragThreshold({
  required DrawPoint from,
  required DrawPoint to,
  required double threshold,
}) {
  if (threshold <= 0) {
    return true;
  }
  final thresholdSquared = threshold * threshold;
  return from.distanceSquared(to) >= thresholdSquared;
}

File: packages/snow_draw_rust_engine/src/draw/input/policies/drag_threshold_policy.rs
#![allow(dead_code)]

use crate::draw::types::draw_point::DrawPoint;

/// Returns whether movement from `from` to `to` reaches `threshold`.
///
/// Non-positive thresholds are treated as immediate drag start.
pub fn has_reached_drag_threshold(from: DrawPoint, to: DrawPoint, threshold: f64) -> bool {
    if threshold <= 0.0 {
        return true;
    }

    let threshold_squared = threshold * threshold;
    from.distance_squared(to) >= threshold_squared
}
